ibc/modules/src/core/ics23_commitment/merkle.rs

```diff
@@ -55,8 +55,8 @@ impl<H> From<RawMerkleProof> for MerkleProof<H> {
 			.into_iter()
 			.map(|p| {
 				let mut encoded = Vec::new();
-				prost::Message::encode(&p, &mut encoded).unwrap();
-				prost::Message::decode(&*encoded).unwrap()
+				prost::Message::encode(&p, &mut encoded).expect("Expects valid proof conversion");
+				prost::Message::decode(&*encoded).expect("Expects valid proof conversion")
```

Comment:
isn't this potentially part of the same problem? A message that's not correct, that will fail on the decoding piece. And this will cause panics, which is what we're trying to avoid.

Reply:
The comment above the code block explains it, we are converting between the same struct, they are just defined in separate crates because of how ibc_proto is generated.